pkg/assembler/clients/operations/hasSBOM.graphql

```diff
@@ -50,9 +50,8 @@ query HasSBOMs($filter: HasSBOMSpec!) {
     ...AllHasSBOMTree
   }
 }
-
-query HasSBOMList($filter: HasSBOMSpec!, $after: ID, $first: Int) {
-  HasSBOMList(hasSBOMSpec: $filter, after: $after, first: $first) {
+query HasSBOMList($filter: HasSBOMSpec!, $after: ID, $first: Int, $getIncludedSoftware: Boolean, $getIncludedDependencies: Boolean, $getIncludedOccurrences: Boolean) {
```

Comment:
we can do the filtering based on what the user asks based on the graphQL query but this does not work when we are using it on the client side and we do not have this ability. This is the reason to add the boolean values to modify the output.

Reply:
What are your thoughts on this @mihaimaruseac? Is there a better way to do this?

Reply:
It seems this is returning different nodes, right? For included software it's only `Package` or `Artifact`, whereas for dependencies is `IsDependency` and for occurences is `IsOccurence`. If so, then users can request what they need from the query

Reply:
how?

Reply:
From the schema: https://github.com/guacsec/guac/blob/1a04f1399485a1ada505496ec9445d3b1ef4bc47/pkg/assembler/graphql/schema/hasSBOM.graphql#L67-L69

So a user that only wants the occurences would have something like

```gql
HasSBOMList(...) {
  edges {
    node {
      includedOccurrences   # THIS ONE
         ...
    }
  }
}
```

instead of including everything.

Then we would probably add these booleans on the client side to send the query with only what is relevant

Reply:
(my gql example might not be fully correct now, cannot test myself right now)

Reply:
But how does that work from the client side query? We don't have the ability to specify what the user wants, unless we create more client side queries that filter out the values...

Reply:
> HasSBOMList(...) {
  edges {
    node {
      includedOccurrences   # THIS ONE
         ...
    }
  }
}
Yeah this is what I was thinking, I will change the PR to that.